GMX-Bots/gmx7-detect-profitable-trades/src/agent.spec.ts

```diff
@@ -0,0 +1,448 @@
+    1,
+    1,
+  ]);
+
+  describe("handleTransaction", () => {
+    //empty findings
+
+    it("returns empty findings if there are no swap events to a non-router address", async () => {
+      const transaction: TransactionEvent = new TestTransactionEvent()
+        .setFrom(createAddress("0x0f"))
+        .setTo(createAddress("0x0f"));
+      const findings = await handler(transaction);
+
+      expect(findings).toStrictEqual([]);
+    });
+
+    it("returns empty findings if there are no swap events to a router address", async () => {
+      const transaction: TransactionEvent = new TestTransactionEvent()
+        .setFrom(createAddress("0x0f"))
+        .setTo(MOCK_GMX_ROUTER_ADDRESS);
+      const findings = await handler(transaction);
+
+      expect(findings).toStrictEqual([]);
+    });
+
+    it("returns empty findings if there are swap events to a non-router address", async () => {
+      const transaction1: TransactionEvent = new TestTransactionEvent()
+        .setFrom(createAddress("0xf0"))
+        .setTo(createAddress("0x0f"))
+        .setBlock(1)
+        .addEventLog(eventGain.format("sighash"), createAddress("0x0"), logGain.data, ...logGain.topics.slice(1));
+      const findings = await handler(transaction1);
+
+      expect(findings).toStrictEqual([]);
+    });
+  });
+
+  it("returns empty findings if there is an account with a suspicious amount of profitable trades but not above the grace period", async () => {
+    const transaction: TransactionEvent = new TestTransactionEvent()
+      .setFrom(createAddress("0xf0"))
+      .setTo(MOCK_GMX_ROUTER_ADDRESS)
+      .setBlock(1)
+      .addEventLog(eventGain.format("sighash"), createAddress("0x0"), logGain.data, ...logGain.topics.slice(1));
+
+    mockPriceFeed
+      .get()
+      .latestRoundData.mockReturnValueOnce({
+        roundId: 0,
+        answer: 100000000,
+        startedAt: 2,
+        updatedAt: 3,
+        answeredInRound: 4,
+      })
+      .mockReturnValueOnce({
+        roundId: 0,
+        answer: 2000000000000,
+        startedAt: 2,
+        updatedAt: 3,
+        answeredInRound: 4,
+      })
+      .mockReturnValueOnce({
+        roundId: 0,
+        answer: 100000000,
+        startedAt: 2,
+        updatedAt: 3,
+        answeredInRound: 4,
+      })
+      .mockReturnValueOnce({
+        roundId: 0,
+        answer: 2000000000000,
+        startedAt: 2,
+        updatedAt: 3,
+        answeredInRound: 4,
+      })
+      .mockReturnValueOnce({
+        roundId: 0,
+        answer: 100000000,
+        startedAt: 2,
+        updatedAt: 3,
+        answeredInRound: 4,
+      })
+      .mockReturnValueOnce({
+        roundId: 0,
+        answer: 2000000000000,
+        startedAt: 2,
+        updatedAt: 3,
+        answeredInRound: 4,
+      });
+    let findings = await handler(transaction);
+    findings = await handler(transaction);
+    findings = await handler(transaction);
+
+    expect(findings).toStrictEqual([]);
```

Comment:
```suggestion
    let findings = await handler(transaction);
    expect(findings).toStrictEqual([]);
    
    findings = await handler(transaction);
    expect(findings).toStrictEqual([]);
    
    findings = await handler(transaction);
    expect(findings).toStrictEqual([]);
```
Suggestion to test each individual call to the handler. Applicable in other unit tests as well.